### DISCUSSION TITLE
Disable need for HTTPS during development?

### DISCUSSION THREAD

**Discussion by m1212e** [panva/openid-client#716] [Q&A] [answered]
Hi,

with the new major release I get `only requests to HTTPS are allowed` when I try to connect to my local mock oidc issuer that I use during development. Is there a way to disable the need for a secure connection for scenarios like these?

Error reference for better searchability: `OAUTH_HTTP_REQUEST_FORBIDDEN`

**Comment by panva** [comment, marked as answer]
Of course https://github.com/panva/openid-client/tree/main/docs#advanced-configuration

**Comment by ManfredLange** [comment]
In case someone else runs in the observation that the flag `allowInsecureRequests` is marked as deprecated: The reason is to highlight the fact that you shouldn't have to use it.

Instead, use it only in development or other non-TLS environments where you are absolutely certain it is secure enough. The default should be HTTPs, though.
